src/main/java/net/dries007/tfc/compat/wailatop/WailaBlocks.java

```diff
@@ -0,0 +1,502 @@
+
+    @Override
+    public List<String> getWailaBody(ItemStack itemStack, List<String> currenttip, IWailaDataAccessor accessor, IWailaConfigHandler config)
+    {
+        Block b = accessor.getBlock();
+        TileEntity te = accessor.getTileEntity();
+
+        // Mechanics
+        if (b instanceof BlockOreTFC) currenttip = OreTFCBody(itemStack, currenttip, accessor, config);
+        else if (te instanceof TEPlacedItemFlat)
+            currenttip = PlacedItemFlatBody(itemStack, currenttip, accessor, config);
+        else if (te instanceof TEPitKiln) currenttip = PitKilnBody(itemStack, currenttip, accessor, config);
+        else if (te instanceof TECharcoalForge)
+            currenttip = CharcoalForgeBody(itemStack, currenttip, accessor, config);
+        else if (te instanceof TELogPile) currenttip = LogPileBody(itemStack, currenttip, accessor, config);
+        else if (te instanceof TEBarrel) currenttip = BarrelBody(itemStack, currenttip, accessor, config);
+        else if (te instanceof TEBloomery) currenttip = BloomeryBody(itemStack, currenttip, accessor, config);
+        else if (te instanceof TEBlastFurnace)
+            currenttip = BlastFurnaceBody(itemStack, currenttip, accessor, config);
+        else if (te instanceof TEFirePit) currenttip = FirepitBody(itemStack, currenttip, accessor, config);
+            // Crops and Trees
+        else if (te instanceof TECropBase) currenttip = CropSimpleBody(itemStack, currenttip, accessor, config);
+        else if (te instanceof TECropSpreading)
+            currenttip = CropSpreadingBody(itemStack, currenttip, accessor, config);
+        else if (b instanceof BlockFruitTreeLeaves)
+            currenttip = FruitTreeLeavesBody(itemStack, currenttip, accessor, config);
+        else if (b instanceof BlockBerryBush) currenttip = BerryBushBody(itemStack, currenttip, accessor, config);
+        else if (b instanceof BlockFarmlandTFC)
+            currenttip = FarmlandTFCBody(itemStack, currenttip, accessor, config);
+        else if (b instanceof BlockCropDead) currenttip = CropDeadBody(itemStack, currenttip, accessor, config);
+        else if (b instanceof BlockFruitTreeLeaves)
+            currenttip = FruitTreeLeavesBody(itemStack, currenttip, accessor, config);
+
+        return currenttip;
+    }
+
+    @Override
+    public List<String> getWailaTail(ItemStack itemStack, List<String> currenttip, IWailaDataAccessor accessor, IWailaConfigHandler config)
+    {
+        if (config.getConfig("tfc.displayTemp"))
+        {
+            currenttip.add(new TextComponentTranslation("waila.temperature").getFormattedText() + " : " + String.valueOf(Math.round(ClimateTFC.getActualTemp(accessor.getWorld(), accessor.getPosition(), 0))) + new TextComponentTranslation("waila.tempsymbol").getFormattedText());
+        }
+        return currenttip;
+    }
+
+    public List<String> FirepitBody(ItemStack itemStack, List<String> currenttip, IWailaDataAccessor accessor, IWailaConfigHandler config)
```

Comment:
A couple things that are common themes with all the below methods:

 - All these methods need to be changed to adhere to java naming conventions - `lowerCamelCase` for methods. In my opinion they should all be prefixed with `get`, i.e. `getFirepitBody`, `getPitKilnBody`, `getOreStack`, etc.
 - There is also no specific need to include `TFC` in the name, despite it being the name of the tile entity.
 - There should be a consistient access level between all of them, imo `private` makes the most sense.

Reply:
TFC is in the name of getOreTFC as the blockname is BlockOreTFC. So for consistency with other methods I've included it there.

Reply:
Other suggestions have been commited.